### DISCUSSION TITLE
Full example for pytest-based unit test with fastapi-users

### DISCUSSION THREAD

**Discussion by gromdimon** [fastapi-users/fastapi-users#1314] [Q&A] [answered]
Hello,

could anyone point me to a working full example of a repository that demonstrates how to run pytest-based unit tests for an app that uses fastapi-users, maybe even with an sqlalchemy backend?

Thanks

**Comment by frankie567** [comment, marked as answer]
Hey 👋 

What I usually do is to actually use `dependency_overrides` feature of FastAPI: https://fastapi.tiangolo.com/advanced/testing-dependencies/#use-the-appdependency_overrides-attribute

So you can do something like this, assuming you have a FastAPI `app`, and a FastAPI Users dependency `current_active_user`

```
app.dependency_overrides[current_active_user] = lambda: test_user
```

The way you create `test_user` depends on you, it can probably be a Pytest fixture which creates a `User` object and may persist in in DB.

**Reply by gromdimon** [reply]
@frankie567 Thank you for a quick answer! Everything's clear now :)

**Comment by galkinc** [comment]
@gromdimon Hi! 
Could you share your example please.

**Reply by gromdimon** [reply]
Hey @galkinc ! 
You can find examples in this project: https://github.com/bihealth/reev
To be more precise here is one of the particular tests:
https://github.com/bihealth/reev/blob/d86e4fed8164f996c478f4d70d7ab9b54fb06ebd/backend/tests/api/api_v1/test_acmgseqvar.py#L41

Hope, this help!
